### PR TITLE
[wxCocoa] Implement wxSYS_CURSOR_[XY] in wxSystemSettings::GetMetric() function.

### PR DESCRIPTION
**Pull request by solemnwarning**

### CODE REVIEW

**Review by vadz — COMMENTED** [review]
Thanks, I'll merge this too, unless @csomor has any comments, just as with the other PR.

File: src/osx/cocoa/settings.mm [hunk 1]
```diff
@@ -221,6 +221,20 @@ static int wxOSXGetUserDefault(NSString* key, int defaultValue)
 // system metrics/features
 // ----------------------------------------------------------------------------
 
+static float GetCursorScale()
+{
+    /* https://github.com/godotengine/godot-proposals/issues/3842#issuecomment-1020553533 */
```
[the remark this answers is not in the document]
Comment (solemnwarning) [reply src/osx/cocoa/settings.mm:226]: Somehow I didn't find that page when researching this... maybe 1AM isn't the best time to be writing macOS code. Will amend later.

File: src/osx/cocoa/settings.mm [hunk 2]
```diff
@@ -233,15 +247,18 @@ static int wxOSXGetUserDefault(NSString* key, int defaultValue)
...
-        // TODO case wxSYS_CURSOR_Y:
         // TODO case wxSYS_DCLICK_X:
         // TODO case wxSYS_DCLICK_Y:
         // TODO case wxSYS_DRAG_X:
         // TODO case wxSYS_DRAG_Y:
         // TODO case wxSYS_EDGE_X:
         // TODO case wxSYS_EDGE_Y:
 
+        case wxSYS_CURSOR_X:
+            return (int)((float)([[[NSCursor arrowCursor] image] size].width) * GetCursorScale());
```
[the remark this answers is not in the document]
Comment (solemnwarning) [reply src/osx/cocoa/settings.mm:258]: Rounding may be more correct - do we typically round when converting macOS floating-point co-ordinates?

**Review by csomor — APPROVED** [review]
looks good to me as well, thanks a lot

**Pull request closed without merging**